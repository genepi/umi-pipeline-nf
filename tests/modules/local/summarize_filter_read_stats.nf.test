nextflow_process {

    name "Test Process SUMMARIZE_FILTER_READ_STATS"
    script "modules/local/umi_processing/summarize_filter_read_stats.nf"
    process "SUMMARIZE_FILTER_READ_STATS"

    test("Should summarize stats for one sample and one target") {

        when{
            process {
                """
                input[0] = Channel.of(tuple(
                    "sample1",
                    "target1",
                    [file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target1_stats1.tsv"),
                    file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target1_stats2.tsv")]
                ))
                input[1] = "raw"
                input[2] = file("$baseDir/bin/summarize_split_read_stats.py", checkIfExists: true)
                """
            }
        }
        
        then{
            assert process.success
            assert snapshot(process.out.summary_stats).match()
        }
    }

    test("Should summarize stats for multiple samples and targets") {

        when{
            process {
                """
                input[0] = Channel.of(tuple(
                    "sample1",
                    "target1",
                    [file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target1_stats1.tsv"),
                    file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target1_stats2.tsv")]
                ),
                tuple(
                    "sample1",
                    "target2",
                    [file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target2_stats1.tsv"),
                    file("${baseDir}/tests/input/summarize_filter_read_stats/sample1_target2_stats2.tsv")]
                ),
                tuple(
                    "sample2",
                    "target2",
                    [file("${baseDir}/tests/input/summarize_filter_read_stats/sample2_target2_stats1.tsv"),
                    file("${baseDir}/tests/input/summarize_filter_read_stats/sample2_target2_stats2.tsv")]
                ))
                input[1] = "raw"
                input[2] = file("$baseDir/bin/summarize_split_read_stats.py", checkIfExists: true)
                """
            }
        }
        
        then{
            assert process.success
            assert snapshot(process.out.summary_stats).match()
        }
    }

}