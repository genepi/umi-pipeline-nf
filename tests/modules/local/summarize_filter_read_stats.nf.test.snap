{
    "Should summarize stats for multiple samples and targets": {
        "content": [
            [
                [
                    "sample1",
                    "target1",
                    "sample1_target1_filtered_reads.tsv:md5,baadf2eeda323dbbbc66f0c9edd0fbed"
                ],
                [
                    "sample1",
                    "target2",
                    "sample1_target2_filtered_reads.tsv:md5,90953c0242123bcc53777746a6fb0d9f"
                ],
                [
                    "sample2",
                    "target2",
                    "sample2_target2_filtered_reads.tsv:md5,004710eae85329cf3c850d5e23f9af1d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-10-07T11:29:05.214066045"
    },
    "Should summarize stats for one sample and one target": {
        "content": [
            [
                [
                    "sample1",
                    "target1",
                    "sample1_target1_filtered_reads.tsv:md5,baadf2eeda323dbbbc66f0c9edd0fbed"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-10-07T11:28:56.116812084"
    }
}